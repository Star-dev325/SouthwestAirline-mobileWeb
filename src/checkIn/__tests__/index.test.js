jest.mock('@swa-ui/encryption', () => ({
  EncryptionProvider: jest.fn().mockReturnValue({ EncryptionProvider: 'MockEncryptionProvider' }),
  useHref: jest.fn().mockReturnValue({ href: 'mock_href' })
}));

import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import CheckIn from 'src/checkIn';

jest.mock('src/checkIn/pages/checkInLandingPage', ()=> () => <div>View CheckIn Page</div>);

describe('CheckIn', () => {
  it('should render CheckIn route page', () => {
    const { container } = createComponent();

    expect(container).toMatchSnapshot();
  });
});

const createComponent = (props = {}) => {
  const defaultProps = {
    match: {
      url: '/check-in'
    }
  };
  const mergedProps = { ...defaultProps, ...props };
  const store = configureMockStore()({
    app: {
      errorHeader: {
        errorMessage: null,
        hasError: false
      }
    },
    router: {
      location: {
        search: 'search'
      }
    }
  });

  return render(
    <Provider store={store}>
      <BrowserRouter>
        <CheckIn {...mergedProps} />
      </BrowserRouter>
    </Provider>
  );
};
