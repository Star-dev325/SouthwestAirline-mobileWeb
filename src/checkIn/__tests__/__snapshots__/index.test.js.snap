// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckIn should render CheckIn route page 1`] = `
<div>
  <div
    class="check-in"
  >
    <div>
      View CheckIn Page
    </div>
  </div>
</div>
`;
